merkletree/batchedproof.go

```diff
@@ -0,0 +1,103 @@
+package merkletree
+
+import "github.com/filecoin-project/go-data-segment/util"
+
+// BatchedMerkleProof represents a Merkle proof of a sequence of leafs
+type BatchedMerkleProof interface {
+	// LeftProof returns the underlying, full, proof of the left-most element proven in the batch
+	LeftProof() MerkleProof
+	// RightProof returns the underlying, full, proof of the right-most element proven in the batch
+	RightProof() MerkleProof
+	// ValidateSequence ensures the correctness of the proof of a sequence of subtrees against the root of a Merkle tree
+	ValidateSequence(firstSubtree *Node, secondSubtree *Node, root *Node) bool
+	// ValidateLeafs ensures the correctness of the proof of a sequence of leafs against a Merkle tree.
+	// startIdx indicates the index in the tree of the left-most leaf contained in the sequence leafs
+	ValidateLeafs(leafs [][]byte, startIdx int, tree MerkleTree) bool
+}
+
+type BatchedProofFactory func() BatchedMerkleProof
+
+type BatchedProofData struct {
+	// leftPath contains the path needed to verify the left-most node only
+	leftPath []Node
+	// rightPath contains the path needed to verify the right-most node only
+	rightPath []Node
+	// commonPath contains the path needed to verify everything between the left-most and right-most nodes
+	commonPath []Node
+	// leftLvl indicates the level in the Merkle tree where the left-most node is located. Root has level 0
+	leftLvl int
+	// leftIdx indicates the index within leftLvl where the left-most node whose membership to prove is located.
+	// Indexing starts at 0
+	leftIdx int
+	// rightLvl indicates the level in the Merkle tree where the right-most node is located. Root has level 0
+	rightLvl int
+	// rightIdx indicates the index within rightLvl where the right-most node whose membership to prove is located.
+	rightIdx int
+}
+
+func CreateEmptyBatchedProof() BatchedMerkleProof {
+	return BatchedProofData{}
+}
```

Comment:
In Go unless there is required initialisation, constructor functions of empty objects are usually not explicit.
You would use `BatchedProofData{}` in its place.

Reply:
Handled in commit [2c743c5](https://github.com/filecoin-project/go-data-segment/pull/2/commits/2c743c5933894afb0cb2b31006a043b860869fa4)